src/json-schema-version.cpp.in

```diff
@@ -0,0 +1,44 @@
+#include <nlohmann/json-schema-version.hpp>
+
+namespace nlohmann
+{
+namespace json_schema
+{
+namespace about
+{
+
+const std::string &version_txt()
+{
+	// clang-format off
+	static std::string version{"${${PROJECT_NAME}_VERSION}"};
+	// clang-format on
+	return version;
+}
+
+// clang-format off
+template <>
+bool check_library_version<${${PROJECT_NAME}_VERSION_MAJOR}>() noexcept
```

Comment:
Yes, I see for clang-format. Just clang-format off the whole file then.

Reply:
> The version.[ch]pp.in -files could be completely generic and with two or three defines (or only one? `JSON_SCHEMA_VALIDATOR_API`) outside their inclusion could be completely independent.

I don't get it. What you mean with "generic"?
The down-side of defines is, that they are potentially subject of name clashes since they are always in public scope. We use the approach of `<SOME-NS>::about>` to have the version info sorted by namespace and libraries. That is our interpretation of generic.

Reply:
I'm trying out my idea to make it clearer for you to understand what I mean.